pages/task/read/routers/read.js

```diff
@@ -104,11 +104,20 @@ module.exports = () => {
   app.use((req, res, next) => {
     const action = req.task.data.action;
     const model = req.task.data.model;
-    if (model === 'establishment') {
-      res.locals.static.values = req.establishment;
-      return next();
-    }
     if (action === 'update' || action === 'delete' || action === 'update-conditions') {
+      // if task is resolved, get previous values from task
+      if (req.task.status === 'resolved') {
```

Comment:
This should check if the task is open rather than a specific status, otherwise the diffs will be wonky for non-resolved closed tasks.

I believe there's a decorator that sets a property like `isOpen` on the task.

Reply:
if they are non-resolved then the data should be looked up from the model, it's only resolved/autoresolved we want, but autoresolved won't be in the task list